src/foam/nanos/crunch/RemoveJunctionsOnUserRemoval.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+foam.CLASS({
+  package: 'foam.nanos.crunch',
+  name: 'RemoveJunctionsOnUserRemoval',
+
+  documentation: 'Rule to remove any user-capability relationships when a user is removed',
+
+  implements: [
+    'foam.nanos.ruler.RuleAction'
+  ],
+
+  javaImports: [
+    'foam.core.ContextAgent',
+    'foam.core.X',
+    'foam.dao.DAO',
+    'foam.nanos.auth.User',
+    'foam.nanos.crunch.UserCapabilityJunction',
+    'static foam.mlang.MLang.*'
+  ],
+
+  methods: [
+    {
+      name: 'applyAction',
+      javaCode: `
+      final Long userId = ((User) obj).getId();
+
+      agency.submit(x, new ContextAgent() {
+        @Override
+        public void execute(X x) {
+          ((DAO) x.get("userCapabilityJunctionDAO"))
```

Comment:
Is there a relationship on on the User to their capabilities that would make this easier?